### PR TITLE
Implement max instruction limit

### PR DESCRIPTION
**Pull request by lofcz**
Exposed in ScriptOptions.InstructionLimit, defaults to 0 = ignored. Changed several related longs to ulongs.

### CODE REVIEW

File: src/WattleScript.Interpreter/Execution/VM/Processor/Processor_InstructionLoop.cs [hunk 1]
```diff
@@ -49,6 +49,11 @@ private DynValue Processing_Loop(int instructionPtr, bool canAwait = false)
 
 					++executedInstructions;
 
+					if (m_Script.Options.InstructionLimit > 0 && executedInstructions > m_Script.Options.InstructionLimit)
+					{
+						throw new ScriptRuntimeException($"Maximum limit of {m_Script.Options.InstructionLimit} exceed. Script terminated.");
```
Comment (CallumDev) [inline src/WattleScript.Interpreter/Execution/VM/Processor/Processor_InstructionLoop.cs:54]: Grammar: exceed -> exceeded

File: src/WattleScript.Tests/EndToEnd/CLikeTestRunner.cs [hunk 2]
```diff
@@ -45,6 +45,7 @@ public async Task RunCore(string path, bool reportErrors = false)
         StringBuilder stdOut = new StringBuilder();
 
         Script script = new Script(CoreModules.Preset_HardSandboxWattle | CoreModules.PrototypeTable);
+        script.Options.InstructionLimit = ulong.MaxValue;
```
Comment (CallumDev) [inline src/WattleScript.Tests/EndToEnd/CLikeTestRunner.cs:48]: Why?

Comment (lofcz) [reply src/WattleScript.Tests/EndToEnd/CLikeTestRunner.cs:48]: Tested if it crashed correctly with `while(true) {print("something")}`. Can it stay this way?

File: src/WattleScript.Interpreter/Execution/VM/Processor/Processor_InstructionLoop.cs (hunk 1 above)
Comment (lofcz) [reply src/WattleScript.Interpreter/Execution/VM/Processor/Processor_InstructionLoop.cs]: Fixed in 0f19ff28b9ce3fe1903549f2d9ab63a12c36316c

**Pull request merged**